testsuites/CBLTester/System_test_multiple_device/test_system_testing.py

```diff
@@ -301,7 +317,9 @@ def test_system(params_from_base_suite_setup):
             t.join()
             results = query.query_get_docs_limit_offset(cbl_db, limit=query_limit,
                                                         offset=query_offset)
-            _releaseQueryResults(base_url, results)
+            # Query results do not store in memory for dot net, so no need to release memory for dotnet
+            if(platform.lower() != "net-msft" and platform.lower() != "uwp" and platform.lower() != "xamarin-ios" and platform.lower() != "xamarin-android"):
```

Comment:
I think you can add the if condition in the method utils.release(results), that way you can just add the condition only once

Reply:
We are not supposed to release the objects if we are not storing in memory.  It might not catch issues if we are storing in memory explicitly and if there is a bug and not storing in memory.

Reply:
if we keep all memory related conditions in the  utils.release method it will be easy for future reference and if we see the same issue in new upcoming platform easy to use.